File: androidApp/src/main/java/com/todo/app/android/screens/component/TextBody.kt
package com.todo.app.android.screens.component

import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview

@Composable
fun TextBody(
    text: String,
    textAlign: TextAlign = TextAlign.Start,
    textColor: Color = Color.Black
) {
    Text(
        text = text,
        style = MaterialTheme.typography.bodyLarge.copy(
            color = textColor,
            textAlign = textAlign
        )
    )
}
@Preview(showBackground = true)
@Composable
fun TextSubHeader_Preview() {
    TextBody("Test")
}

File: iosApp/iosApp/screens/components/TextBody.swift
import SwiftUI

struct TextBody: View {
    let text: String
    var textAlign: TextAlignment = .leading
    var textColor: Color = .black
    
    var body: some View {
        Text(text)
            .font(.body) // This corresponds to MaterialTheme.typography.bodyLarge in Compose
            .foregroundColor(textColor) // Sets the text color
            .multilineTextAlignment(textAlign) // Aligns the text (left, center, right)
    }
}

struct TextBody_Previews: PreviewProvider {
    static var previews: some View {
        TextBody(text: "Test")
            .previewLayout(.sizeThatFits) // Adjust the layout for better preview
            .padding() // Add padding for visibility in the preview
    }
}
